src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal class TypeImportCompletionItem : CompletionItem
+    {
+        public int TypeArity { get; }
+
+        public int OverloadCount { get; }
+
+        public string ContainingNamespace { get; }
+
+        private string _metadataName = null;
+        public string MetadataName
+        {
+            get
+            {
+                if (_metadataName == null)
+                {
+                    _metadataName = ComposeAritySuffixedMetadataName(
+                       GetFullyQualifiedName(this.ContainingNamespace, this.DisplayText),
+                       this.TypeArity);
+                }
+
+                return _metadataName;
+            }
+        }
+
+        public object CachedObject { get; set; }
```

Comment:
1. docs.  
2. should we just do this at the CompletionItem level?

Reply:
Sure, I'm fine with that